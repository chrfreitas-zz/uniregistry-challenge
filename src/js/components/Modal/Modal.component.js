import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Modal extends Component {
    static propTypes = {
        show: PropTypes.bool,
        domain: PropTypes.object
    }

    getClass() {
        let className = 'modal';

        if(!this.props.show){
            className += ' hide';
        }

        return className;
    }

    render() {
        return (
            <div className={ this.getClass() }>
                <form action="">
                    <div>
                        <label>Domain name</label>
                        <input type="text"/>
                    </div>
                    <div>
                        <label>Registran email</label>
                        <input type="text"/>
                    </div>
                    <div>
                        <label>Price</label>
                        <input type="text"/>
                    </div>
                    <div>
                        <button>Update it!</button>
                        <a href="">Cancel</a>
                    </div>
                </form>
            </div>
        );
    }
}

export default Modal;