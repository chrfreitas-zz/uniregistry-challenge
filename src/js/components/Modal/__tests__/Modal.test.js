import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Modal from '../Modal.component';

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

describe('Modal component', () => {
    it('should match with the snapshot', () => {
        const domain = {
            id: 2,
            name: 'lalala',
            email: 'lalala@gmail.com',
            price: '$3.00'
        }

        const wrapper = renderer.create(<Modal domain={domain}/>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});