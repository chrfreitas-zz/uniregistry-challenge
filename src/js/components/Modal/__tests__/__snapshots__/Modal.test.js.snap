// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component It should match with the snapshot 1`] = `
<div
  className="modal"
>
  <form
    action=""
  >
    <div>
      <label>
        Domain name
      </label>
      <input
        type="text"
      />
    </div>
    <div>
      <label>
        Registran email
      </label>
      <input
        type="text"
      />
    </div>
    <div>
      <label>
        Price
      </label>
      <input
        type="text"
      />
    </div>
    <div>
      <button>
        Update it!
      </button>
      <a
        href=""
      >
        Cancel
      </a>
    </div>
  </form>
</div>
`;
